libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.cpp

```diff
@@ -419,7 +419,7 @@ bool ESP8266WiFiGenericClass::mode(WiFiMode_t m, WiFiState* state) {
     }
 
     if (wifi_fpm_get_sleep_type() != NONE_SLEEP_T) {
-        // wifi may have been put asleep by ESP8266WiFiGenericClass::preinitWiFiOff
+        // wifi starts asleep by default
```

Comment:
Looking at esp32's ::mode:
https://github.com/espressif/arduino-esp32/blob/46d5afb17fb91965632dc5fef237117e1fe947fc/libraries/WiFi/src/WiFiGeneric.cpp#L541

Should this be more in-line with that?
i.e.
- getMode() checks for fpm -> returns null-mode when sleeping
- mode() consecutively turns off the wifi when WIFI_OFF without an additional call to force sleep and turns it back on as well

Reply:
:+1: thanks !

I added 1) and 3).
I'm reticent about 2) because `WiFi.mode(WIFI_OFF)` will not put the modem to sleep, or puting modem to sleep does not necessarily switches the mode to `WIFI_OFF`. 2) should come with that no ?

Reply:
I meant exactly that, to do forcesleep with WiFi.mode(WIFI_OFF) and resume with any real mode.
Not noticed the special shutdown and resume modes though, I wonder if those should be separate methods instead and not a 'mode'?

Reply:
They are indeed "fake" mode and described as such somewhere. There are two examples making use of them.
They are designed to import the deep sleep procedure into the core and allow users to not use random fw-snippets to put the chip to sleep.

Reply:
Aaah, both are already implemented as methods. No reason examples could not use those directly?